### PR TITLE
Eyedropper: handle rgb string when getting the eyedropper color

### LINKED ISSUE #12560: Eyedropper not working
## Bug Description

On Firefox browser, the eyedropper to pick a color from the canvas loads continuously and does not open the focus circle. This occurs anywhere the eyedropper color picker is available. 

## Expected Behaviour

The eyedropper should allow the user to select a color from the canvas on Firefox browser.

## Steps to Reproduce

1. Add a new story on Firefox browser
2. Add an element
3. Click on eyedropper
4. Note endless load

## Screenshots
Current Behavior on Firefox:

https://user-images.githubusercontent.com/66372350/199023978-edf65069-aaca-4f4c-b453-2e870300a6d7.mov

Expected Behavior (on Chrome):

https://user-images.githubusercontent.com/66372350/198608683-44bd27b9-9a3d-45e8-8338-7a164a3e0370.mov

## Additional Context
Support topic: https://wordpress.org/support/topic/infinite-loop-on-color-picker-arrow-in-firefox/

- Plugin Version: 1.26.0
- WordPress Version:
- Operating System: macOS
- Browser: Firefox

### PR DESCRIPTION
**Pull request by miina**
## Relevant Technical Choices
- Adds handling colour string in rgb.
- Removes part of the patch that breaks the logic for FF and Safari.

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Add a text element (or any other that has styling with colors)
2. Use eyedropper for changing color: test in the colour picker, Style tab and also in the Floating edit menu of the element.
3. Verify eyedropper works as expected.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #12560

### CODE REVIEW

**Comment by swissspidy** [comment]
Did the `window.EyeDropper` implementation change or what happened here? [According to the spec](https://wicg.github.io/eyedropper-api/#colorselectionresult-dictionary) the eyedropper result is supposed to always be a hex string.

Also, not sure if it was clear on the issue: there was a similar thing reported at #12681 / https://github.com/GoogleForCreators/web-stories-wp/issues/12681#issuecomment-1314081435 for Chrome, and #12560 mentioned issues on Firefox

**Comment by miina** [comment]
> Did the window.EyeDropper implementation change or what happened here? [According to the spec](https://wicg.github.io/eyedropper-api/#colorselectionresult-dictionary) the eyedropper result is supposed to always be a hex string.

Yep, saw that as well but in practice it can be a rgb string 🤷 Didn't find out why it was happening. Thoughts?

> Also, not sure if it was clear on the issue: there was a similar thing reported at https://github.com/GoogleForCreators/web-stories-wp/issues/12681 / https://github.com/GoogleForCreators/web-stories-wp/issues/12681#issuecomment-1314081435 for Chrome, and https://github.com/GoogleForCreators/web-stories-wp/issues/12560 mentioned issues on Firefox

Alright, I think I might have confused the issues, checking FF now.

**Comment by miina** [comment]
@swissspidy as far as I can see [here](https://developer.mozilla.org/en-US/docs/Web/API/EyeDropper_API) then the EyeDropper API is not supported in FF nor Safari (I can confirm the same behaviour in Safari as in FF).
In the code as well, `isEyeDropperApiSupported` gives `false` in these browsers.

Looks like something changed meanwhile? 🤔 I can see from the original issue that @merapi [had tested on all supported browsers](https://github.com/GoogleForCreators/web-stories-wp/pull/7961#issuecomment-866280254).

Thoughts?

**Comment by timarney** [comment]
Took a quick look as I was curious and worked on the EyeDropper API PR

Found a few things that might be of interest 
Here's a similar report (forGnome Linux devices)
https://github.com/WICG/eyedropper-api/issues/28

Also 

If you watch the video in this tweet 
https://twitter.com/matanbobi/status/1432681072790171656 

You'll see the selected output


Screen Shot 2022-11-15 at 8 01 46 AM


If you visit the codepen linked from that tweet and test  https://codepen.io/matanbobi/pen/GREoZvo 

The output format is different 
```
Selected color is rgb(221, 201, 125)
```

Screen Shot 2022-11-15 at 8 02 54 AM

File: patches/html-to-image+1.10.8.patch [hunk 1]
```diff
@@ -1,25 +1,5 @@
...
-     await new Promise((resolve, reject) => {
--        clonedNode.onload = resolve;
-         clonedNode.onerror = reject;
-         if (clonedNode instanceof HTMLImageElement) {
-             clonedNode.srcset = '';
-@@ -30,6 +29,7 @@ async function embedImageNode(clonedNode, options) {
-         else {
-             clonedNode.href.baseVal = dataURL;
-         }
-+        clonedNode.decode().then(() => resolve(clonedNode)).catch(reject);
```
Comment (miina) [inline patches/html-to-image+1.10.8.patch:17]: This was added within #12623 but it seems to break the logic for FF and Safari. Looks like removing this doesn't influence the Karma tests (the same tests are still failing/flaky as on `main`), so not sure exactly why this was added for the tests.

Comment (barklund) [reply patches/html-to-image+1.10.8.patch:17]: It's pretty confusing to read a diff of a diff 😂

**Review by barklund — APPROVED** [review]
If this is what it takes... 🤷

**Comment by kkalarickal** [comment]
@miina - I successfully tested the color picker in the Style panel and the floating menu, via the eyedrop icon and inside the color picker box - for the following actions on these different browser combinations:

- Color picker for text, text box (fill & highlight) 
- Color picker for GIF / Image Filter
- Color picker for Shapes
- Page background

Browsers:

Win10 - Firefox, Chrome, Edge
Mac - Safari, Chrome, Firefox

ISSUE: I could NOT get the Color Picker EyeDropper to open on Opera (Mac or Win)
 

Pls let me know if I should open another ticket for Opera EyeDropper issue. CC: @swissspidy

**Comment by swissspidy** [comment]
@kkalarickal Yes please!

**Pull request merged**